File: src/go/dating.go
package main

import (
	"fmt"
	"time"
)

func getAgeInYears(birthday time.Time) int {
	today := time.Now()
	result := today.Year() - birthday.Year()
	if today.YearDay() < birthday.YearDay(){
		result--
	}
	return result
}

func main(){
	birthday := time.Date(1939, time.October, 27, 0, 0, 0, 0, time.UTC)
	age := getAgeInYears(birthday)
	fmt.Printf("John Cleese was born in %s and is %d years old.\n", birthday.Format(time.RFC3339)[:10], age)

	parsedDate, _ := time.Parse("2006-01-02T15:04:05", "1975-11-10T01:25:00");
	fmt.Printf("Parsed date is %s", parsedDate.Format(time.RFC3339)[:19])
}

File: src/rb/dating.rb
require "date"

def GetAgeInYears(birthday)
    today = DateTime.now
    return today.year - birthday.year - (today.yday < birthday.yday ? 1 : 0)
end

birthday = Date.new(1939, 10, 27)
age = GetAgeInYears(birthday)
puts "John Cleese was born in #{birthday.strftime("%F")} and is #{age} years old."

parsedDate = DateTime.parse("1975-11-10T01:25:00")
puts "Parsed date is #{parsedDate.strftime("%FT%T")}"